apps/chat-e2e/src/tests/sharedPromptIcons.test.ts

```diff
@@ -1,204 +1,359 @@
+    setTestIds(
+      'EPMRTC-1517',
+      'EPMRTC-1817',
+      'EPMRTC-1520',
+      'EPMRTC-1521',
+      'EPMRTC-1518',
+      'EPMRTC-3156',
+      'EPMRTC-1523',
+    );
     let prompt: Prompt;
+    let shareLinkResponse: ShareByLinkResponseModel;
+
     await dialTest.step('Prepare a new prompt', async () => {
       prompt = promptData.prepareDefaultPrompt();
       await dataInjector.createPrompts([prompt]);
     });
 
     await dialTest.step(
-      'Open prompt dropdown menu and choose "Share" option',
+      'Open prompt dropdown menu, choose "Share" option and verify modal window text',
       async () => {
         await dialHomePage.openHomePage();
         await dialHomePage.waitForPageLoaded({
           isNewConversationVisible: true,
         });
         await prompts.openEntityDropdownMenu(prompt.name);
-        await promptDropdownMenu.selectMenuOption(MenuOptions.share);
+        await prompts.shareEntity();
+        await shareModalAssertion.assertMessageContent(
+          ExpectedConstants.sharePromptText,
+        );
       },
     );
 
     await dialTest.step(
       'Click on "Cancel" button in modal window and verify no shared icon appears on prompt icon',
       async () => {
         await shareModal.closeButton.click();
-        await expect
-          .soft(
-            prompts.getEntityArrowIcon(prompt.name),
-            ExpectedMessages.promptIsNotShared,
-          )
-          .toBeHidden();
+        await promptAssertion.assertEntityArrowIconState(
+          { name: prompt.name },
+          'hidden',
+        );
       },
     );
 
     await dialTest.step(
       'Open Share modal again, click outside modal window area and verify no shared icon appears on prompt icon',
       async () => {
         await prompts.openEntityDropdownMenu(prompt.name);
-        await promptDropdownMenu.selectMenuOption(MenuOptions.share);
+        await prompts.shareEntity();
         // eslint-disable-next-line playwright/no-force-option
         await promptBar.draggableArea.click({ force: true });
-        await expect
-          .soft(
-            shareModal.getElementLocator(),
-            ExpectedMessages.modalWindowIsClosed,
-          )
-          .toBeHidden();
+        await shareModalAssertion.assertModalState('hidden');
+        await promptAssertion.assertEntityArrowIconState(
+          { name: prompt.name },
+          'hidden',
+        );
+      },
+    );
 
-        await expect
-          .soft(
-            prompts.getEntityArrowIcon(prompt.name),
-            ExpectedMessages.promptIsNotShared,
-          )
-          .toBeHidden();
+    await dialTest.step(
+      'Open Share modal again, copy link with keyboard and verify no shared icon appears',
+      async () => {
+        await prompts.openEntityDropdownMenu(prompt.name);
+        await prompts.shareEntity();
+        await shareModal.shareLinkInput.click();
+        await page.keyboard.press(keys.ctrlPlusA);
```

Comment:
I think that we need to introduce a method to select and copy text to the clipboard, since we have pasteFromClipboard and copyToClipboard methods

Reply:
agree, fixed